File: thrift-python-benchmark/py/NonBlockingServer.py
import sys
import movie_service_impl
import logging

from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol
from thrift.protocol.TBinaryProtocol import TBinaryProtocolAccelerated
from thrift.server import TServer
from thrift.server import TNonblockingServer
reload(sys)
sys.setdefaultencoding('utf-8')

from movieservice import MovieService

class PingHandler:
    def __init__(self):
        self.text = "Hello!"

    def ping(self):
        print("ping()")


if __name__ == '__main__':
    logger = logging.getLogger('thrift.server.TNonblockingServer')
    logger.setLevel(logging.DEBUG)
    # create file handler which logs even debug messages
    fh = logging.FileHandler('thrift_python_server.log')

    fh.setLevel(logging.INFO)
    # create console handler with a higher log level
    ch = logging.StreamHandler()
    ch.setLevel(logging.INFO)  # handler = PingHandler()
    # create formatter and add it to the handlers
    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
    fh.setFormatter(formatter)
    ch.setFormatter(formatter)
    # add the handlers to the logger
    logger.addHandler(fh)
    logger.addHandler(ch)

    handler = movie_service_impl.MovieServiceAsyncHandler()

    processor = MovieService.Processor(handler)
    # processor = Ping.Processor(handler)
    transport = TSocket.TServerSocket(host="127.0.0.1", port=9090)

    t_factory = TBinaryProtocol.TBinaryProtocolFactory()
    p_factory = TBinaryProtocol.TBinaryProtocolAcceleratedFactory()
    server = TNonblockingServer.TNonblockingServer(processor, transport, t_factory, p_factory, threads=100)

    print 'Starting server'
    server.serve()

File: thrift-java-benchmark/src/main/java/thrift/benchmark/NonBlockingServer.java
package thrift.benchmark;

import movieservice.MovieService;
import movieservice.Movies;
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.server.THsHaServer;
import org.apache.thrift.server.TNonblockingServer;
import org.apache.thrift.server.TServer;
import org.apache.thrift.server.TThreadPoolServer;
import org.apache.thrift.transport.TNonblockingServerSocket;
import org.apache.thrift.transport.TNonblockingServerTransport;
import org.apache.thrift.transport.TTransportException;

import java.io.File;
import java.io.IOException;
import java.nio.file.Paths;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.logging.FileHandler;
import java.util.logging.Level;
import java.util.logging.Logger;
import java.util.logging.SimpleFormatter;

/**
 * Created by thuy on 21/05/16.
 */
public class NonBlockingServer {
    private static final java.util.logging.Logger logger = Logger.getLogger(NonBlockingServer.class.getName());
    public static MovieServiceImpl handler;
    public static MovieService.Processor processor;
    public static int port;
    public static TServer server;
    private FileHandler fh;

    public NonBlockingServer(int port) throws IOException, TTransportException {
        this.port = port;
        Movies moviesData = MovieServiceUtil.parseMovies(MovieServiceUtil.getDefaultMoviesFile());
        handler = new MovieServiceImpl(moviesData);
        processor = new MovieService.Processor(handler);
        try {
            TNonblockingServerTransport serverTransport = new TNonblockingServerSocket(port);
            server = new THsHaServer(new THsHaServer.Args(serverTransport).
                    processor(processor));

            SimpleDateFormat format = new SimpleDateFormat("MM_dd_yyyy_HHmmss");

            String dir = Paths.get("").toAbsolutePath().toString() + "//Logging";
            File directory = new File(dir);

            if (!directory.exists()) {
                directory.mkdir();
            }

            fh = new FileHandler(dir + "//Thrift_Server_NON_Blocking_"
                    + format.format(Calendar.getInstance().getTime()) + ".log");
        } catch (TTransportException e) {
            e.printStackTrace();
            logger.log(Level.SEVERE, "TTransport exception -", e);
        } catch (Exception e) {
            e.printStackTrace();
            logger.log(Level.SEVERE, "Unknown exception -", e);

        }
        fh.setFormatter(new SimpleFormatter());
        fh.setLevel(Level.INFO);
        logger.addHandler(fh);

    }

    public void start() {
        logger.info("Starting the NON server...");
        logger.info("Server listening on port: " + port);
        server.serve();
    }

}
